paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/action/CloneActionITCase.java

```diff
@@ -0,0 +1,400 @@
+        writeDataToDynamicBucket(
+                Arrays.asList(
+                        rowData(24, 898, 1, BinaryString.fromString("20221209")),
+                        rowData(12, 312, 90, BinaryString.fromString("20221208")),
+                        rowData(29, 434, 15, BinaryString.fromString("20221209"))),
+                2);
+        checkLatestSnapshot(table, 2, Snapshot.CommitKind.APPEND);
+
+        /** do clone */
+        runCloneLatestSnapshotAction();
+        FileStoreTable targetTable = getFileStoreTable(targetTableName);
+        checkLatestSnapshot(targetTable, 2, Snapshot.CommitKind.APPEND);
+
+        /** check scan result */
+        List<DataSplit> splits1 = table.newSnapshotReader().read().dataSplits();
+        assertThat(splits1.size()).isEqualTo(6);
+        List<DataSplit> splits2 = targetTable.newSnapshotReader().read().dataSplits();
+        assertThat(splits2.size()).isEqualTo(6);
+
+        TableScan sourceTableScan = table.newReadBuilder().newScan();
+        TableScan targetTableScan = targetTable.newReadBuilder().newScan();
+
+        List<String> scanResult =
+                Arrays.asList(
+                        "+I[1, 100, 15, 20221201]",
+                        "+I[12, 312, 90, 20221208]",
+                        "+I[2, 40000, 16, 20221208]",
+                        "+I[24, 898, 1, 20221209]",
+                        "+I[29, 434, 15, 20221209]",
+                        "+I[4, 100, 19, 20221209]");
+        validateResult(table, ROW_TYPE, sourceTableScan, scanResult, 60_000);
+        validateResult(targetTable, ROW_TYPE, targetTableScan, scanResult, 60_000);
+
+        /**
+         * check file name and file size of manifest && index && data && schema && changelog &&
+         * snapshot files
+         */
+        List<Path> targetTableFiles = PickFilesUtil.getUsedFilesForLatestSnapshot(targetTable);
+        List<Pair<Path, Path>> filesPathInfoList =
+                targetTableFiles.stream()
+                        .map(
+                                absolutePath ->
+                                        Pair.of(
+                                                absolutePath,
+                                                getPathExcludeTableRoot(
+                                                        absolutePath, targetTable.location())))
+                        .collect(Collectors.toList());
+
+        Path tableLocation = table.location();
+        for (Pair<Path, Path> filesPathInfo : filesPathInfoList) {
+            Path sourceTableFile = new Path(tableLocation.toString() + filesPathInfo.getRight());
+            Assertions.assertTrue(table.fileIO().exists(sourceTableFile));
+            Assertions.assertEquals(
+                    table.fileIO().getFileSize(sourceTableFile),
+                    targetTable.fileIO().getFileSize(filesPathInfo.getLeft()));
+        }
+    }
+
+    @Test
+    @Timeout(60_000)
+    public void testSnapshotWithSlowExpire() throws Exception {
```

Comment:
Slow expire is safe. No need for such case.

Reply:
done